capture_service/device_mgr.cc

```diff
@@ -509,201 +649,177 @@ absl::Status DeviceManager::DeployReplayApk(const std::string &serial)
-        if (!ret.ok())
-        {
-            LOGW("WARNING: Could not unpin GPU clock: %s\n", std::string(ret.message()).c_str());
-        }
-    }
+absl::Status DeviceManager::RunReplayProfilingBinary(const GfxrReplaySettings &settings) const
+{
+    LOGD("RunReplayProfilingBinary(): SETUP\n");
+    LOGD("RunReplayProfilingBinary(): Deploy libraries and binaries\n");
+    std::string copy_cmd = absl::StrFormat("push %s %s",
+                                           ResolveAndroidLibPath(kProfilingPluginFolderName, "")
+                                           .string(),
+                                           kTargetPath);
+    RETURN_IF_ERROR(m_device->Adb().Run(copy_cmd));
+    std::string remote_profiling_dir = absl::StrFormat("%s/%s",
+                                                       kTargetPath,
+                                                       kProfilingPluginFolderName);
 
-    ret = m_device->Adb().Run("shell setprop compositor.high_priority 1");
-    if (!ret.ok())
+    std::string binary_path_on_device = absl::StrFormat("%s/%s",
+                                                        remote_profiling_dir,
+                                                        kProfilingPluginName);
+    RETURN_IF_ERROR(m_device->Adb().Run(absl::StrCat("shell chmod +x ", binary_path_on_device)));
+
+    LOGD("RunReplayProfilingBinary(): RUN\n");
+    std::string metrics_str = absl::StrJoin(settings.metrics, " ");
+    std::string gfxr_replay_flag = settings.replay_flags_str.empty() ?
+                                   "" :
+                                   absl::StrFormat("--gfxr_replay_flags \\\"%s\\\"",
+                                                   settings.replay_flags_str);
+    std::string cmd = absl::StrFormat("shell %s %s %s %s",
+                                      binary_path_on_device,
+                                      settings.remote_capture_path,
+                                      gfxr_replay_flag,
+                                      metrics_str);
+    // TODO(b/449174476): Remove this redundant statement when the command is logged before it hangs
+    LOGD("Profiling binary cmd: %s\n", cmd.c_str());
+    RETURN_IF_ERROR(m_device->Adb().Run(cmd));
+
+    LOGD("RunReplayProfilingBinary(): RETRIEVE ARTIFACTS\n");
+    std::filesystem::path parse_remote_path = settings.remote_capture_path;
+    std::string           csv_local_name = absl::StrFormat("%s%s",
+                                                 parse_remote_path.stem(),
+                                                 kProfilingMetricsCsvSuffix);
+    std::string csv_remote_file_path = parse_remote_path.replace_extension(".csv").string();
+
+    if (absl::Status s = m_device->RetrieveFile(csv_remote_file_path,
+                                                settings.local_download_dir,
+                                                true,
+                                                csv_local_name);
```

Comment:
nit: consider using [function argument comments](https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments) to clarify what `true` means in this context:
```suggestion
    if (absl::Status s = m_device->RetrieveFile(csv_remote_file_path,
                                                settings.local_download_dir,
                                                /*delete_after_retrieve=*/true,
                                                csv_local_name);
```